fe1-web/src/features/witness/components/WitnessNotification.tsx

```diff
@@ -0,0 +1,106 @@
+import PropTypes from 'prop-types';
+import React, { FunctionComponent, useEffect, useMemo } from 'react';
+import { Text, View } from 'react-native';
+import { Button } from 'react-native-elements';
+import { useSelector } from 'react-redux';
+
+import { dispatch } from 'core/redux';
+
+import { WitnessHooks } from '../hooks';
+import { WitnessFeature, MESSAGE_TO_WITNESS_NOTIFICATION_TYPE } from '../interface';
+import { requestWitnessMessage } from '../network/WitnessMessageApi';
+import { makeMessageToWitnessSelector, removeMessageToWitness } from '../reducer';
+
+const WitnessNotification = ({ notification, navigateToNotificationScreen }: IPropTypes) => {
+  const messageSelector = useMemo(
+    () => makeMessageToWitnessSelector(notification.messageId),
+    [notification.messageId],
+  );
+  const message = useSelector(messageSelector);
+
+  const discardNotifications = WitnessHooks.useDiscardNotifications();
+  const markNotificationAsRead = WitnessHooks.useMarkNotificationAsRead();
+  const isEnabled = WitnessHooks.useIsEnabled();
+  const laoId = WitnessHooks.useCurrentLaoId();
+
+  const onWitness = () => {
+    if (message) {
+      dispatch(
+        discardNotifications({ laoId: laoId.valueOf(), notificationIds: [notification.id] }),
+      );
+      if (isEnabled) {
+        requestWitnessMessage(message.channel, message.message_id);
+      }
+      dispatch(removeMessageToWitness(message.message_id.valueOf()));
+      navigateToNotificationScreen();
+    }
+  };
+
+  const onDecline = () => {
+    if (message) {
+      dispatch(markNotificationAsRead({ laoId: laoId.valueOf(), notificationId: notification.id }));
+      navigateToNotificationScreen();
+    }
+  };
+
+  // if the notification state somehow gets out of sync, remove the corresponding notification
+  useEffect(() => {
+    if (!message) {
+      dispatch(
+        discardNotifications({ laoId: laoId.valueOf(), notificationIds: [notification.id] }),
+      );
+      navigateToNotificationScreen();
+    }
+  }, [laoId, navigateToNotificationScreen, discardNotifications, notification.id, message]);
```

Comment:
logically, I'd expect this to appear before the `onWitness`/ `onDecline` definitions.
Also, is this just a safeguard ? When is the "somehow gets out of sync" expected ?
If it's just added for robustness, I'd log debugging information to make sure that we know and understand what happens (and we aren't just redirected to some other page).

Reply:
Makes sense!

Yes it is just as a safeguard :) I will add debugging information!